### PR TITLE
Convert 'server closed idle connection' to errDiskNotFound

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
In some corner cases, when the client is sending a POST request but 
the server decides a keep-alive timeout situation, io.EOF will be sent 
to the client. Golang retries the request in some cases about avoid to 
do that for a POST request with a body already written to the server 
(to avoid making a change in the server twice)

In such unfortunate use case, MinIO will simplify the situation as a 
networking issue and will mark the disk offline for a short period.

## Motivation and Context
Avoid decomissioning quitting for a retriable situation

## How to test this PR?
Hard to test

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by klauspost — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

File: cmd/storage-rest-client.go [hunk 1]
```diff
@@ -50,9 +50,18 @@ func isNetworkError(err error) bool {
 		return xnet.IsNetworkOrHostDown(nerr.Err, false)
 	}
 
+	switch err.Error() {
```
Comment (klauspost) [inline cmd/storage-rest-client.go]: Maybe `strings.Contains(err.Error(), "server closed idle connection")` - http likes to wrap errors.

Comment (vadmeste) [reply cmd/storage-rest-client.go]: > Maybe strings.Contains(err.Error(), "server closed idle connection") - http likes to wrap errors.

Yes, done.. this is not a hot path anyway

**Review by klauspost — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by tdsn0007** [comment]
Hi, I think my problem will be solved with this pull. I have Veeam costumer using my minio S3, and the problem is timout in Veeam application when they start to send data.


Feb  7 15:27:18 s3-t0-02 minio[802158]: API: SYSTEM(bucket=costumer, object=Veeam/Archive/Veeam/4ce89a68-aa47-4ad2-b8af-679f0dbc0c85/00000000-0000-0000-0000-000000000000/blocks/e66eb04d26da1cc7306fb6ea98b13dc9/896606.9a1ab664094d03204a3de68334fa33b9.5d1f76e28dd873be864b091d3b622a40.blk)
Feb  7 15:27:18 s3-t0-02 minio[802158]: Time: 20:27:18 UTC 02/07/2023
Feb  7 15:27:18 s3-t0-02 minio[802158]: DeploymentID: 70afc619-df38-4c4a-94a0-904306b92d28
Feb  7 15:27:18 s3-t0-02 minio[802158]: Error: Drive http://s3-t0-01.s3site.com:9000/hdd-04, path (costumer/Veeam/Archive/Veeam/4ce89a68-aa47-4ad2-b8af-679f0dbc0c85/00000000-0000-0000-0000-000000000000/blocks/e66eb04d26da1cc7306fb6ea98b13dc9/896606.9a1ab664094d03204a3de68334fa33b9.5d1f76e28dd873be864b091d3b622a40.blk) returned an error (Post "http://s3-t0-01.s3site.com:9000/minio/storage/hdd-04/v49/readversion?disk-id=b2ac1cc3-793f-4371-967a-e1f66efeafc4&file-path=Veeam%2FArchive%2FVeeam%2F4ce89a68-aa47-4ad2-b8af-679f0dbc0c85%2F00000000-0000-0000-0000-000000000000%2Fblocks%2Fe66eb04d26da1cc7306fb6ea98b13dc9%2F896606.9a1ab664094d03204a3de68334fa33b9.5d1f76e28dd873be864b091d3b622a40.blk&read-data=false&version-id=0ed78cd2-737b-4901-b113-0a97a7cbee13&volume=costumer": http: server closed idle connection) (*fmt.wrapError)
Feb  7 15:27:18 s3-t0-02 minio[802158]:       7: internal/logger/logger.go:258:logger.LogIf()
Feb  7 15:27:18 s3-t0-02 minio[802158]:       6: internal/logger/logonce.go:104:logger.(*logOnceType).logOnceIf()
Feb  7 15:27:18 s3-t0-02 minio[802158]:       5: internal/logger/logonce.go:135:logger.LogOnceIf()
Feb  7 15:27:18 s3-t0-02 minio[802158]:       4: cmd/erasure-metadata-utils.go:187:cmd.readAllFileInfo()
Feb  7 15:27:18 s3-t0-02 minio[802158]:       3: cmd/erasure-healing.go:1033:cmd.erasureObjects.HealObject()
Feb  7 15:27:18 s3-t0-02 minio[802158]:       2: cmd/erasure-sets.go:1209:cmd.(*erasureSets).HealObject()



In Veeam, the error is timout. Did you know if this pull will solve my problem?

**Comment by marcosalberto** [comment]
Hi @tdsn0007, i hope you are fine.

Have you managed to fix this issue after updating MinIO?

Thanks in advance.

**Pull request merged**